src/versions.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable */
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// Note: Leave this file as ES5 js for compatibility with earlier Node.js versions
+'use strict';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const pjson = require('../package.json');
+
+/**
+ * Determines whether or not a tag string is a semantic version.
+ *
+ * @param {*} tag The possible version string
+ * @returns {boolean} True, if the string is recognized as a semantic version
+ */
+export function isVersion(tag) {
+  if (!tag) {
+    return false;
+  }
+  // From https://github.com/sindresorhus/semver-regex
+  const SEMVER_REGEX =
+    /^v?(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-[\da-z-]+(?:\.[\da-z-]+)*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?$/gi;
+  return SEMVER_REGEX.test(tag.toString());
+}
+
+module.exports.isVersion = isVersion;
+
+/**
+ * Compares two semantic version strings.
+ *
+ * @param {string} a The first version
+ * @param {string} b The second version
+ * @returns {number} < 0 if a < b, 0 if a == b, > 0 if a > b
+ */
+export function compareVersions(a, b) {
```

Comment:
Should we use the semver library instead?

Reply:
I will look into replacing with semver.